pkg/statistics/handle/autoanalyze/priorityqueue/calculatoranalysis/calculator_analysis_test.go

```diff
@@ -0,0 +1,356 @@
+	sortPrioritiesByWeight(newPriorities)
+
+	// Convert newPriorities to CSV string
+	var buf bytes.Buffer
+	writer := csv.NewWriter(&buf)
+	writer.Write([]string{"ID", "CalculatedPriority", "TableSize", "Changes", "TimeSinceLastAnalyze", "ChangeRatio"})
+	for _, p := range newPriorities {
+		writer.Write([]string{
+			fmt.Sprintf("%d", p.job.ID),
+			fmt.Sprintf("%.4f", p.priority),
+			fmt.Sprintf("%.0f", p.job.TableSize),
+			fmt.Sprintf("%.0f", p.job.Changes),
+			fmt.Sprintf("%.0f", p.job.TimeSinceLastAnalyze),
+			fmt.Sprintf("%.4f", p.changeRatio),
+		})
+	}
+	writer.Flush()
+	got := buf.Bytes()
+
+	// Compare with golden file
+	golden := goldenFile
+	if *update {
+		t.Log("updating golden file")
+		err = os.WriteFile(golden, got, 0644)
+		require.NoError(t, err, "Failed to update golden file")
+	}
+
+	want, err := os.ReadFile(golden)
+	require.NoError(t, err, "Failed to read golden file")
+
+	require.Equal(t, want, got)
+}
+
+// calculateNewPriorities calculates priorities for all jobs
+func calculateNewPriorities(jobs []TestJob, calculator *priorityqueue.PriorityCalculator) []jobWithPriority {
+	newPriorities := make([]jobWithPriority, 0, len(jobs))
+	for _, job := range jobs {
+		priority := calculator.CalculateWeight(&job)
+		changeRatio := job.Changes / job.TableSize // Calculate change ratio
+		newPriorities = append(newPriorities, jobWithPriority{job, priority, changeRatio})
+	}
+	return newPriorities
+}
+
+// sortPrioritiesByWeight sorts jobs by priority in descending order
+func sortPrioritiesByWeight(priorities []jobWithPriority) {
+	sort.Slice(priorities, func(i, j int) bool {
+		return priorities[i].priority > priorities[j].priority
+	})
+}
+
+// generateCombinations generates test data combinations for analyzing table statistics.
+//
+// The function creates combinations based on different table sizes and time intervals
+// since the last analysis. It calculates reasonable change amounts for each combination
+// using the following principles:
+//
+//  1. Maximum Change Rate: Assumes a maximum change rate of 0.1% of the table size per second.
+//     This rate can be adjusted based on real-world observations.
+//
+//  2. Change Calculation: For each table size and time interval, it calculates a maximum
+//     reasonable change as: tableSize * maxChangeRate * timeSinceLastAnalyze.
+//
+//  3. Change Limit: The maximum change is capped at 300% of the table size to prevent
+//     unrealistic scenarios for very long time intervals.
+//
+//  4. Change Variety: Generates six different change amounts for each combination:
+//     1%, 10%, 50%, 100%, 200%, and 300% of the calculated maximum change.
+//     This variety allows testing the priority calculator under different degrees of modification.
+//
+//  5. Realistic Constraints: Only includes changes that are greater than zero and
+//     do not exceed 300% of the table size, ensuring all data points are meaningful for testing.
```

Comment:
Looks like 3 and 5 actually mean the same thing?
And it may need to be changed after we remove the 1000-row restriction, right? Because it will easily exceed 300% when a table grow from 10 rows to 100 rows.

Reply:
Make sense. Updated.